sdk/src/test/java/dev/cdevents/CDEventsSpecExamplesTest.java

```diff
@@ -0,0 +1,1026 @@
+package dev.cdevents;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.cdevents.constants.CDEventConstants;
+import dev.cdevents.events.*;
+import dev.cdevents.models.CDEvent;
+import dev.cdevents.models.testcaserun.finished.Content;
+import dev.cdevents.models.testcaserun.finished.TestCase;
+import dev.cdevents.models.testcaserun.queued.Trigger;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CDEventsSpecExamplesTest {
+
+    private static String EXAMPLES_FOLDER = CDEventConstants.SPEC_REPO + File.separator + "examples";
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+
+    @Test
+    void testArtifactPackagedCDEvent() throws IOException {
+
+        File artifactPackagedExample = new File(EXAMPLES_FOLDER + File.separator + "artifact_packaged.json");
+        JsonNode expectedNode = objectMapper.readTree(artifactPackagedExample);
+        String expectedJson = objectMapper.writeValueAsString(expectedNode);
+        CDEvent expectedCDEvent = CDEvents.cdEventFromJson(expectedJson);
+        ArtifactPackagedCDEvent expectedEvent = (ArtifactPackagedCDEvent) expectedCDEvent;
+
+        ArtifactPackagedCDEvent createdEvent =  new ArtifactPackagedCDEvent();
+        createdEvent.setSource(URI.create("/event/source/123"));
+        createdEvent.setSubjectId("pkg:golang/mygit.com/myorg/myapp@234fd47e07d1004f0aed9c");
+        createdEvent.setSubjectSource(URI.create("/event/source/123"));
+        createdEvent.setSubjectChangeId("myChange123");
+        createdEvent.setSubjectChangeSource("my-git.example/an-org/a-repo");
```

Comment:
Except for this section, these tests can also be generated or generalised eventually

Reply:
Will do a refactor of these tests in a separate PR